onnxruntime/core/optimizer/selectors_actions/selector_action_transformer.h

```diff
@@ -81,21 +83,22 @@ class SelectorActionRegistry {
   void RegisterSelectorAndAction(const std::string& name,
                                  const OpVersionsMap& ops_and_versions,
                                  std::unique_ptr<NodeSelector> selector,
-                                 std::unique_ptr<Action> action);
+                                 std::unique_ptr<Action> action,
+                                 const std::string& domain = kOnnxDomain);
```

Comment:
I'd think that the domain should be part of the specification of the ops that a selector/action entry applies to (until now this specification was `ops_and_versions`). then we could have a single selector/action entry that can handle ops from multiple domains.

Reply:
The Action object is tied to a specific domain due to the API
https://github.com/microsoft/onnxruntime/blob/ad511bc757e50d16d043d8d45056a36c8140f848/onnxruntime/core/optimizer/selectors_actions/actions.h#L133

Reply:
Each Action object is tied to a specific Domain.

Reply:
Not quite following. The RuntimeState is provided as an argument, and the Domain method is overridable. Why can't the implementation of a specific Action that needs to be dynamic based on the RuntimeState's target node do that?

Reply:
I guess you are asking why cant we use one Action and keep changing the domain as we work through different items?
I think due to the class hierarchy and base class implementing some functionality, it is error prone.  Making that change requires modifying base class(es). I think it is cleaner creating another object.

Reply:
Ed and I discussed this further (as we've done the most work on the select/action setup) and the consensus was to not push the domain check up to this level unless we're going to switch to using OpIdentifier for the checks. Doing so limits the functionality by restricting the transformer to a single domain which is undesirable. 

For now it should be fine to add the internal NHWC domain to the initial check (domain is kOnnxDomain or kMSDomain) and use the existing code.

The selector can add any logic required around ops in the internal NHWC domain. That should be minimal as there are limited EPs that request NHWC and the only other one using L2 optimizers is the CUDA EP. From a look at the L2 optimizers only the ones involving Conv could potentially have a node in the NHWC domain. 

SelectorActionTransformer is only used for L2 and above optimizers as its intended for usage in a minimal build where L1 happens during creation of the ORT format model.

Reply:
The domain (if it is not kOnnxDomain) now specified as a part of the operator in the OpsVersionsMap.